### PR TITLE
Add documentation for using DataConnectionService [SUP-523]

### PR DESCRIPTION
**Pull request by frant-hartm**
Related platform PR https://github.com/hazelcast/hazelcast-mono/pull/3533

### CODE REVIEW

**Comment by frant-hartm** [comment]
@oliverhowell @amandalindsay Before jumping on this one, it might be worth waiting to have devs/QE to review it first - I am not 100 % sure it's all understandable.

File: docs/modules/data-connections/pages/data-connection-service.adoc [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+created.
+
+== Data Connection Sharing
+
+Data connection is shared by default. It means that when the data connection is requested in multiple places, the same
+underlying resource (e.g. Jdbc pool, remote client) is used.
+If you want to share the data connection configuration, but use a different instance of the underlying resource,
+set the `DataConnectionConfig#setShared` to false.
+
+== Typical Usage
```
[the remark this answers is not in the document]
Comment (frant-hartm) [reply docs/modules/data-connections/pages/data-connection-service.adoc]: 5991a9b404a51ab618c1ea2a22d5f71dcf2cdc2a

File: docs/modules/data-connections/pages/how-to-map-loader-data-connection.adoc [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+
+== Before you begin
+
+To complete this tutorial, you need the following:
+
+[cols="1a,1a"]
+|===
+|Prerequisites|Useful resources
+
+|Java !! current java version !!
```
[the remark this answers is not in the document]
Comment (frant-hartm) [reply docs/modules/data-connections/pages/how-to-map-loader-data-connection.adoc]: 5991a9b404a51ab618c1ea2a22d5f71dcf2cdc2a

**Review by Rob-Hazelcast — CHANGES_REQUESTED** [review]
These look like well-structured procedures with prerequisites, numbered steps etc. There are a handful of bugs and inconsistencies, and a few bits I'd like to see fleshed out for improved usability.

File: docs/modules/data-connections/pages/data-connection-service.adoc [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+try (Connection connection = jdbcDataConnection.getConnection()) {
+    // ... work with connection
+} catch (SQLException e) {
+    throw new RuntimeException("Failed to load value for key=" + key, e);
+}
+
+jdbcDataConnection.release();
+----
+
+== Retrieve Data Connection Service
```
[the remark this answers is not in the document]
Comment (frant-hartm) [reply docs/modules/data-connections/pages/data-connection-service.adoc:32]: Yes, these sections provide more details.

File: docs/modules/data-connections/pages/how-to-map-loader-data-connection.adoc [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+}
+----
+
+When you run this class you should see the following output:
+
+[source,text]
+----
+1 maps to one
+42 maps to null
+----
```
[the remark this answers is not in the document]
Comment (frant-hartm) [reply docs/modules/data-connections/pages/how-to-map-loader-data-connection.adoc:273]: 245266b

File: docs/modules/data-connections/pages/how-to-pipeline-service-data-connection.adoc [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+----
+
+== Step 3. the Pipeline and the Job
+
+Create a class `EnrichUsingDataConnection` with `main` method that will be the main job class. The `main` method will do the following steps:
+
+- configure data connection
+- create a service and mapping function using the data connection
+- create a pipeline that
+-- takes a sequence of numbers
```
[the remark this answers is not in the document]
Comment (frant-hartm) [reply docs/modules/data-connections/pages/how-to-pipeline-service-data-connection.adoc]: 330acb1

File: docs/modules/data-connections/pages/data-connection-service.adoc [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+underlying resource (e.g. Jdbc pool, remote client) is used.
+If you want to share the data connection configuration, but use a different instance of the underlying resource,
+set the `DataConnectionConfig#setShared` to false.
+
+== Configuration Considerations
+
+If the data connection is defined in the Hazelcast configuration, it remains immutable for the entire lifespan of the Hazelcast member. In this case, whether you retrieve the DataConnection instance once or each time before accessing the underlying resource, the result will be the same.
+
+However, if the data connection is created dynamically via SQL, it can be replaced using `CREATE OR REPLACE DATA CONNECTION` (see xref:sql
+.adoc). In such cases, the DataConnection instance will stay valid until you release it, allowing you to retrieve the underlying resource as needed. This approach can be useful for adapting to changes in data connection configuration.
```
[the remark this answers is not in the document]
Comment (Rob-Hazelcast) [reply docs/modules/data-connections/pages/data-connection-service.adoc]: Still broken. I don't see a sql.adoc in the repo. I think you want create-data-connection.adoc.

File: docs/modules/data-connections/pages/data-connection-service.adoc [hunk 7]
```diff
@@ -0,0 +1,71 @@
+= Using Data Connections in custom components
+:description: Using the Data Connection Service gives access to the configured data connections in custom components.
```
[the remark this answers is not in the document]
Comment (Rob-Hazelcast) [reply docs/modules/data-connections/pages/data-connection-service.adoc]: Not if you don't add them to the right place in nav.adoc

**Review by Rob-Hazelcast — COMMENTED** [review]
Looking much better, just a couple of outstanding issues. Most important is to reference the new files in nav.adoc. It would be nice to have a validation rule for this @oliverhowell!

**Review by ollyhowell — APPROVED** [review]

**Review by TomaszGaweda — APPROVED** [review]

**Review by Rob-Hazelcast — APPROVED** [review]

**Pull request merged**